rotkehlchen/globaldb/handler.py

```diff
@@ -839,46 +862,176 @@ def get_evm_tokens(
         - Protocol for which to return tokens
         - ignore_spam (default True) to filter out "spam protocol" assets
         """
+        return GlobalDBHandler._get_tokens_from_db(
+            table_name='evm_tokens',
+            token_deserializer=EvmToken.deserialize_from_db,
+            token_type_name='EvmToken',
+            exceptions=exceptions,
+            protocol=protocol,
+            ignore_spam=ignore_spam,
+            chain_filter=('chain', chain_id.serialize_for_db()),
+            needs_underlying_tokens=True,
+            extra_columns=['chain'],
+        )
+
+    @staticmethod
+    def _get_single_token_from_db(
+            table_name: str,
+            token_deserializer: Callable,
+            token_type_name: str,
+            address_filter: tuple[str, Any],
+            chain_filter: tuple[str, Any] | None = None,
+            needs_underlying_tokens: bool = False,
+            extra_columns: list[str] | None = None,
+    ) -> SolanaToken | EvmToken | None:
+        """Generic helper to retrieve a single token by address from database.
+
+        Args:
+            table_name: Database table to query
+            token_deserializer: Function to deserialize database entries
+            token_type_name: Name for error logging
+            address_filter: Tuple of (column, value) for address filtering
+            chain_filter: Optional tuple of (column, value) for chain filtering
+            needs_underlying_tokens: Whether to fetch underlying tokens
+            extra_columns: Additional columns to select
+
+        """
+        # Build SELECT columns
+        base_columns = [
+            'A.identifier', 'B.address', 'B.token_kind', 'B.decimals', 'C.name',
+            'A.symbol', 'A.started', 'A.swapped_for', 'A.coingecko', 'A.cryptocompare', 'B.protocol',  # noqa: E501
+        ]
+        if extra_columns:
+            base_columns[2:2] = [f'B.{col}' for col in extra_columns]
```

Comment:
is this better or just to use the insert method?

Reply:
i think this is better. `.insert()` needs a for loop